modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -423,18 +449,263 @@ public Response blacklistApplications(@HeaderParam("Accept") String acceptHeader
+		if (log.isDebugEnabled()) {
+			log.debug("Invoking notification operation");
+		}
+
+		MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
+		Message message = new Message();
+
+		try {
+			Notification notification = notificationBeanWrapper.getOperation();
+
+			if (notification == null) {
+				throw new OperationManagementException("Notification bean is empty");
+			}
+
+			ProfileOperation operation = new ProfileOperation();
+			operation.setCode(AndroidConstants.OperationCodes.NOTIFICATION);
+			operation.setType(Operation.Type.PROFILE);
+			operation.setPayLoad(notification.toJSON());
+
+			return AndroidAPIUtils.getOperationResponse(notificationBeanWrapper.getDeviceIDs(),
+					operation, message, responseMediaType);
+
+		} catch (OperationManagementException e) {
+			message.setResponseMessage("Issue in retrieving operation management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (DeviceManagementException e) {
+			message.setResponseMessage("Issue in retrieving device management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (OperationConfigurationException e) {
+			message.setResponseMessage("Issue in setting up payload in operation");
+			throw  new AndroidOperationException(message, responseMediaType);
+		}
+	}
+
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("wifi")
+	public Response configureWifi(@HeaderParam("Accept") String acceptHeader,
+								  WifiBeanWrapper wifiBeanWrapper) {
+		if (log.isDebugEnabled()) {
+			log.debug("Invoking configure wifi operation");
+		}
+
+		MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
+		Message message = new Message();
+
+		try {
+			Wifi wifi = wifiBeanWrapper.getOperation();
+
+			if (wifi == null) {
+				throw new OperationManagementException("Wifi bean is empty");
+			}
+
+			ProfileOperation operation = new ProfileOperation();
+			operation.setCode(AndroidConstants.OperationCodes.WIFI);
+			operation.setType(Operation.Type.PROFILE);
+			operation.setPayLoad(wifi.toJSON());
+
+			return AndroidAPIUtils.getOperationResponse(wifiBeanWrapper.getDeviceIDs(),
+					operation, message, responseMediaType);
+
+		} catch (OperationManagementException e) {
+			message.setResponseMessage("Issue in retrieving operation management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (DeviceManagementException e) {
+			message.setResponseMessage("Issue in retrieving device management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (OperationConfigurationException e) {
+			message.setResponseMessage("Issue in setting up payload in operation");
+			throw  new AndroidOperationException(message, responseMediaType);
+		}
+	}
+
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("encrypt")
```

Comment:
better if the path name can be changed to a noun instead of a verb (i.e. encrypt) - as REST APIs normally represent resources